manager/state/raft/raft.go

```diff
@@ -126,6 +124,8 @@ type Node struct {
 
 	snapshotInProgress chan uint64
 	asyncTasks         sync.WaitGroup
+
+	stopped chan struct{}
```

Comment:
I don't think we need to add the `stopped` channel. Maybe the goroutine that waits for this channel to be closed could instead select on `<-n.stopCh` and `<-n.removeRaftCh`?